src/auth/auth.service.ts

```diff
@@ -33,4 +38,28 @@ export class AuthService {
 
     return this.userService.create(userDto);
   }
+
+  async login(user: JwtUser) {
+    const token = this.jwtService.sign(user);
+
+    return {
+      accessToken: token,
+      statusCode: HttpStatus.CREATED,
+    };
+  }
+
+  async validateUser(username: string, pass: string) {
+    const user: UserEntity | null = await this.userService.findOne(username);
+
+    if (user !== null) {
+      const passwordCorrect = bcrypt.compareSync(pass, user.password);
+      const userData = { email: user.email, id: user.id };
+
+      if (passwordCorrect) {
+        return userData;
+      }
+    }
+
+    return null;
```

Comment:
should better follows the return early pattern, it could improve readability of code.

e.g.
```typescript
if (!user) return null;

const passwordCorrect = ...
```

Reply:
Revised.
[refactor: revised recommend in auth.service.ts](https://github.com/Cophr/backend/pull/10/commits/a7106295d99711d88471d9a81f22865281b49e3b)